source/13_envs/dmc2gym.rst

```diff
@@ -0,0 +1,336 @@
+
+-  The size of the observed image can be adjusted by setting the \ ``height, width``\ parameters.
+
+-  Observe the specific shape of the observation space by setting \ ``channels_first``\ .
+
+   -  \ ``channels_first=True``\ , observation shape is \ ``[3, height, width]``\
+
+   -  \ ``channels_first=False``\ ，observation shape is \ ``[height, width, 3]``\
+
+-  The range of a single pixel value for each channel is \ ``[0, 255]``\ ， and the data type is \ ``uint8``\
+
+Non-Image Based Observation 
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+-  when \ ``from_pixels=False``\ is set，the viewing space dimension follows \ ``dim(O)``\ in the above table
+
+-  The default range is \ ``[-inf, inf]``\ 
+
+Action Space
+--------
+
+-  Action space dimensions follow \ ``dim(A)``\ in the above table
+
+-  dmc2gym normalizes the action space, the range of each dimension is \ ``[-1, 1]``\ ，the type is \ ``float32``\ .
+
+Reward Space
+--------
+
+Image Based Observation 
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+-  It is related to the \ ``frame_skip``\ parameter，which means that each step is based on the image of \ ``frame_skip``\ frame ，and the dimension is \ ``(1,)``\
+
+- The range is \ ``[0, frame_skip]``\ ，the type is \ ``float32``\ ，default \ ``frame_skip = 1``\
+
+   - The reward space of each frame is \ ``[0, 1]``\ ，and the reward of \ ``frame_skip``\ are superimposed together as the overall reward
+
+Non-Image Based Observation 
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- The dimension is \ ``(1, )``\ ，the range is  \ ``[0, 1]``\ ，the type is \ ``float32``\ .
+
+Other
+====
+
+Abort Condition
+--------
+
+Control tasks are divided into finite-horizon, firstexit and infinite-horizon. DeepMind Control Suite belongs to infinite-horizon, so there is no aborted state or time limit for tasks. 
+
+Lazy Initialization
+----------
+
+In order to facilitate parallel operations such as environment vectorization, environment instances generally implement lazy initialization ，which means \ ``__init__``\ method does not initialize the real original environment instance，but only sets relevant parameters and configuration values. The concrete original environment instance is initialized when the \ ``reset``\ method is used.
+
+
+Random Seeds
+--------
+
+-  There are two parts of random seeds in the environment that need to be set, one is the random seed of the original environment, and the other is the random seed of the random library used by various environment transformations（such as \ ``random``\ ，\ ``np.random``\)
+
+-  For the environment caller, just set two seeds through the\ ``seed``\ method of the environment，and do not need to care about the specific implementation details.
+
+-  Concrete implementation inside the environment: For the seed of the original environment, set before calling the \ ``reset``\ method of the environment ，before the concrete \ ``reset``\ 
```

Comment:
set before calling the \ ``reset``\ method of the environment ，before the concrete \ ``reset``\ -> set in the reset methods of the environment calling function, before the concrete environment implementation reset